jerry-core/lit/lit-strings.h

```diff
@@ -41,8 +107,27 @@ bool lit_is_utf8_string_valid (const lit_utf8_byte_t *, lit_utf8_size_t);
 
 /* iteration */
 lit_utf8_iterator_t lit_utf8_iterator_create (const lit_utf8_byte_t *, lit_utf8_size_t);
-ecma_char_t lit_utf8_iterator_read_code_unit_and_increment (lit_utf8_iterator_t *);
-bool lit_utf8_iterator_reached_buffer_end (const lit_utf8_iterator_t *);
+
+void lit_utf8_iterator_set_to_bos (lit_utf8_iterator_t *);
+void lit_utf8_iterator_set_to_eos (lit_utf8_iterator_t *);
+
+lit_utf8_iterator_pos_t lit_utf8_iterator_get_pos (const lit_utf8_iterator_t *);
+void lit_utf8_iterator_restore_pos (lit_utf8_iterator_t *, lit_utf8_iterator_pos_t);
+
+ecma_length_t lit_utf8_iterator_get_index (const lit_utf8_iterator_t *);
+
+ecma_char_t lit_utf8_iterator_read_next (const lit_utf8_iterator_t *);
+ecma_char_t lit_utf8_iterator_read_prev (const lit_utf8_iterator_t *);
+
+void lit_utf8_iterator_incr (lit_utf8_iterator_t *);
+void lit_utf8_iterator_decr (lit_utf8_iterator_t *);
+void lit_utf8_iterator_advance (lit_utf8_iterator_t *, ecma_length_t);
+
+ecma_char_t lit_utf8_iterator_read_next_and_incr (lit_utf8_iterator_t *);
+ecma_char_t lit_utf8_iterator_read_prev_and_decr (lit_utf8_iterator_t *);
```

Comment:
read usually means decode and consume. I would prefer

lit_utf8_iterator_read_next_and_incr -> lit_utf8_iterator_read_next

peek means do not consume:
lit_utf8_iterator_peek_next -> lit_utf8_iterator_read_next